### ISSUE TITLE
Coverage Minimum causing errors

### ISSUE THREAD

**Issue by cdenneen** [rodjek/rspec-puppet#397]
Recently the code coverage report with minimum coverage percentage started failing:

``` ruby
RSpec.configure do |c|
  c.after(:suite) do
    RSpec::Puppet::Coverage.report!(95)
  end
end
```

Causes failures like so:

```
/tmp/profile/.bundle/gems/rspec-core-3.1.7/lib/rspec/core/example_group.rb:444:in `run': wrong number of arguments (0 for 1) (ArgumentError)
        from /tmp/profile/.bundle/bundler/gems/rspec-puppet-79d5373dcbfa/lib/rspec-puppet/coverage.rb:85:in `coverage_test'
        from /tmp/profile/.bundle/bundler/gems/rspec-puppet-79d5373dcbfa/lib/rspec-puppet/coverage.rb:74:in `report!'
        from /tmp/profile/spec/spec_helper.rb:32:in `block (2 levels) in <top (required)>'
        from /tmp/profile/.bundle/gems/rspec-core-3.1.7/lib/rspec/core/example.rb:315:in `instance_exec'
        from /tmp/profile/.bundle/gems/rspec-core-3.1.7/lib/rspec/core/example.rb:315:in `instance_exec_with_rescue'
        from /tmp/profile/.bundle/gems/rspec-core-3.1.7/lib/rspec/core/hooks.rb:357:in `run'
        from /tmp/profile/.bundle/gems/rspec-core-3.1.7/lib/rspec/core/hooks.rb:427:in `block in run'
        from /tmp/profile/.bundle/gems/rspec-core-3.1.7/lib/rspec/core/hooks.rb:427:in `each'
        from /tmp/profile/.bundle/gems/rspec-core-3.1.7/lib/rspec/core/hooks.rb:427:in `run'
        from /tmp/profile/.bundle/gems/rspec-core-3.1.7/lib/rspec/core/hooks.rb:500:in `run'
        from /tmp/profile/.bundle/gems/rspec-core-3.1.7/lib/rspec/core/runner.rb:113:in `block in run_specs'
        from /tmp/profile/.bundle/gems/rspec-core-3.1.7/lib/rspec/core/reporter.rb:53:in `report'
        from /tmp/profile/.bundle/gems/rspec-core-3.1.7/lib/rspec/core/runner.rb:107:in `run_specs'
        from /tmp/profile/.bundle/gems/rspec-core-3.1.7/lib/rspec/core/runner.rb:85:in `run'
        from /tmp/profile/.bundle/gems/rspec-core-3.1.7/lib/rspec/core/runner.rb:69:in `run'
        from /tmp/profile/.bundle/gems/rspec-core-3.1.7/lib/rspec/core/runner.rb:37:in `invoke'
        from /tmp/profile/.bundle/gems/rspec-core-3.1.7/exe/rspec:4:in `<main>'
/opt/rubies/2.1.8/bin/ruby -I/tmp/profile/.bundle/gems/rspec-core-3.1.7/lib:/tmp/profile/.bundle/gems/rspec-support-3.1.2/lib /tmp/profile/.bundle/gems/rspec-core-3.1.7/exe/rspec --pattern spec/\{classes,defines,unit,functions,hosts,integration,types\}/\*\*/\*_spec.rb --color failed
```

Removing the percentage causes no errors:

``` ruby
RSpec.configure do |c|
  c.after(:suite) do
    RSpec::Puppet::Coverage.report!
  end
end
```

Gemfile using 

``` ruby
  gem "rspec-puppet", :git => 'https://github.com/rodjek/rspec-puppet.git'
```

Puppetversion `4.3.2`
Was hoping @beergeek might have some insight

**Comment by cdenneen** [comment]
I have a test module to show the issue with Percentage. Used it for another PUP bug for the #364 issue but it can be used to show the coverage percentage error here:
https://github.com/cdenneen/profile

**Comment by jantman** [comment]
From what I can tell, this is an rspec version issue. I see this also, using rspec 3.1.0 and rspec-core 3.1.7.

The relevant line, as pointed to by the error message, is [lib/rspec-puppet/coverage.rb L85](https://github.com/rodjek/rspec-puppet/blob/f51dc974a2d5c0cbd6469c4cd4ff0882619593e8/lib/rspec-puppet/coverage.rb#L85), ``coverage_test.run``.

In rspec-core <= 3.2.0, the Rspec::Core::ExampleGroup.run method took a required argument for a reporter. [This commit](https://github.com/rspec/rspec-core/commit/81da012a152d59e432159b93fdd7f48424d642f2) released in 3.2.0 set NullReporter as a default for that.

For me, upgrading to rspec >= 3.2.0 fixes this.
